src/Game.java

```diff
@@ -48,15 +52,23 @@ public int takeATurn(Player whoseTurn){
 		boolean keepGoing = true;
 		printStartRoundMessage(whoseTurn);
 		while(keepGoing){
-			int roll = die.nextInt(7);
+			//int roll = die.nextInt(7);
+			//this line can return any number from 0 to 6. needs to be from 1 to 6.
+			int roll = die.nextInt(6) + 1;
 			String spin = spinner.spin();
 			System.out.println(roll+ " "+ spin);
 			
 			if(roll == LOSER_ROLL){
 				System.out.println("Lose a turn.");
+				if(spin.equals(LOSER_SPIN.toUpperCase())) {
+					System.out.println("Too bad!  Lose all your points.");
+					whoseTurn.resetScore();
+				}
 				return 0;
 			}
-			else if(spin == LOSER_SPIN.toUpperCase()){
+			//else if(spin == LOSER_SPIN.toUpperCase()){
+			//spin is an object type not a primitive type so .equals should be used instead of ==.
+			else if(spin.equals(LOSER_SPIN.toUpperCase())) {
```

Comment:
Your fix introduces some copied code, though.  Better to rearrange the logic than to copy it.